airflow/www/static/js/dag/details/SetDagTaskNotes.tsx

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  Box,
+  Button,
+  Text,
+  Textarea,
+} from '@chakra-ui/react';
+import ResizeTextarea from 'react-textarea-autosize';
+import { getMetaValue } from '../../utils';
+import { useSetDagRunNotes, useSetTaskInstanceNotes } from '../../api';
+
+interface Props {
+  dagId: string;
+  runId: string;
+  taskId: string | undefined;
+  mapIndex: number | undefined;
+  initialValue: string | undefined | null;
+}
+
+const canEdit = getMetaValue('can_edit') === 'True';
+
+const SetDagTaskNotes = ({
+  dagId, runId, taskId, mapIndex, initialValue,
+}: Props) => {
+  const [notes, setNotes] = useState(initialValue == null ? '' : initialValue);
+  const [noteBeforeEdit, setNoteBeforeEdit] = useState('');
+  const [editMode, changeEditMode] = useState(false);
+  const tiStr = (taskId != null ? taskId : '');
+  const {
+    mutateAsync: apiCallToSetDagRunNote, isLoading: dagRunIsLoading,
+  } = useSetDagRunNotes(dagId, runId, notes);
+  const {
+    mutateAsync: apiCallToSetTINote, isLoading: tiIsLoading,
+  } = useSetTaskInstanceNotes(dagId, runId, tiStr, (mapIndex != null ? mapIndex : -1), notes);
+
+  const objectIdentifier = (taskId == null) ? 'DAG Run' : 'Task Instance';
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (taskId == null) {
+      await apiCallToSetDagRunNote();
+    } else {
+      await apiCallToSetTINote();
+    }
+    changeEditMode(false);
+  };
+
+  return (
+    <Accordion defaultIndex={[0]} allowMultiple>
```

Comment:
```suggestion
    <Accordion allowToggle>
```
We only have 1 AccordionItem, the `allowMultiple` should not be used I think. [allow multiple ](https://chakra-ui.com/docs/components/accordion#expand-multiple-items-at-once)

Reply:
You are right :)! Should be better now